src/it/scala/com/exasol/cloudetl/kinesis/KinesisShardsMetadataReaderIT.scala

```diff
@@ -0,0 +1,72 @@
+package com.exasol.cloudetl.kinesis
+
+import java.sql.ResultSet
+
+import org.testcontainers.containers.localstack.LocalStackContainer
+
+class KinesisShardsMetadataReaderIT extends KinesisAbstractIntegrationTest {
+  val TEST_STREAM_NAME = "Test_stream"
+
+  override final def beforeAll(): Unit = {
+    prepareContainers()
+    createKinesisStream(TEST_STREAM_NAME, 3)
+    setupExasol()
+    createKinesisMetadataScript()
+  }
+
+  test("returns shards from a stream") {
+    val resultSet = this.executeKinesisMetadataScript("VALUES (('0', '0'))")
+    assert(resultSet.next() === true)
+    assert(resultSet.getString("KINESIS_SHARD_ID") === "shardId-000000000000")
+    assert(resultSet.getString("SHARD_SEQUENCE_NUMBER") === null)
+    assert(resultSet.next() === true)
+    assert(resultSet.getString("KINESIS_SHARD_ID") === "shardId-000000000001")
+    assert(resultSet.getString("SHARD_SEQUENCE_NUMBER") === null)
+    assert(resultSet.next() === true)
+    assert(resultSet.getString("KINESIS_SHARD_ID") === "shardId-000000000002")
+    assert(resultSet.getString("SHARD_SEQUENCE_NUMBER") === null)
+    assert(resultSet.next() === false)
+
+  }
+
+  test("returns shards from a stream with existing metadata in table") {
+    val resultSet = this.executeKinesisMetadataScript(
+      """VALUES (('shardId-000000000000', '1234'),
+        |       ('shardId-000000000001', '5678'),
+        |       ('shardId-000000000004', '9012'))""".stripMargin
+    )
+    assert(resultSet.next() === true)
+    assert(resultSet.getString("KINESIS_SHARD_ID") === "shardId-000000000000")
+    assert(resultSet.getString("SHARD_SEQUENCE_NUMBER") === "1234")
+    assert(resultSet.next() === true)
+    assert(resultSet.getString("KINESIS_SHARD_ID") === "shardId-000000000001")
+    assert(resultSet.getString("SHARD_SEQUENCE_NUMBER") === "5678")
+    assert(resultSet.next() === true)
+    assert(resultSet.getString("KINESIS_SHARD_ID") === "shardId-000000000002")
+    assert(resultSet.getString("SHARD_SEQUENCE_NUMBER") === null)
+    assert(resultSet.next() === false)
+  }
```

Comment:
Can you use the `collectToResultSet` function in these two sets?

Reply:
done